openfisca_core/parameters.py

```diff
@@ -727,17 +709,53 @@ def __getitem__(self, key):
             return result
 
 
+def _parse_child(child_name, child, child_path):
+    if 'values' in child:
+        return Parameter(child_name, child, child_path)
+    elif 'brackets' in child:
+        return Scale(child_name, child, child_path)
+    elif isinstance(child, dict) and all([INSTANT_PATTERN.match(key) for key in child.keys()]):
+        return Parameter(child_name, child, child_path)
+    else:
+        return ParameterNode(child_name, data = child, file_path = child_path)
+
+
 def _compose_name(path, child_name):
     if path:
-        if isinstance(child_name, int)or INSTANT_PATTERN.match(child_name):
+        if isinstance(child_name, int) or INSTANT_PATTERN.match(child_name):
             return '{}[{}]'.format(path, child_name)
         return '{}.{}'.format(path, child_name)
     else:
         return child_name
 
 
+def _validate_parameter(parameter, data, data_type = None, allowed_keys = None):
+    type_map = {
+        dict: 'object',
+        list: 'array',
+        }
+
+    if data_type is not None and not isinstance(data, data_type):
+        raise ParameterParsingError(
+            "'{}' must be of type {}.".format(parameter.name, type_map[data_type]).encode("utf-8"),
+            parameter.file_path
+            )
+
+    if allowed_keys is not None:
+        keys = data.keys()
+        for key in keys:
+            if key not in allowed_keys:
+                raise ParameterParsingError(
+                    "Unexpected property '{}' in '{}'. Allowed properties are {}."
+                    .format(key, parameter.name, list(allowed_keys)).encode('utf-8'),
+                    parameter.file_path
+                    )
+
+
 def contains_nan(vector):
     if np.issubdtype(vector.dtype, np.record):
         return any([contains_nan(vector[name]) for name in vector.dtype.names])
     else:
         return np.isnan(np.min(vector))
+
+
```

Comment:
Extra blank line ?

Reply:
Nop, this one is required by `flake`. The rule is:

- One line betweens class methods
- Two lines betweens independent functionds

Reply:
Arg, I should have made sure `flake` was passing before asking for a review